lib/policyEvaluator/evaluator.js

```diff
@@ -133,11 +133,17 @@ function meetConditions(requestContext, statementCondition, log) {
             // reference_policies_multi-value-conditions.html)
             const keyBasedOnRequestContext =
                 findConditionKey(key, requestContext);
-            // Handle IfExists
+            // Handle IfExists and negation operators
+            const operatorsWithNegation = ['StringNotEquals',
```

Comment:
They are still constructed every time we call the function, that's better than every iteration, but it's still an allocation we'll do on every function call.